lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+const errors = require('internal/errors');
+
+if (!inspector || !inspector.asyncTaskScheduled) return;
+
+module.exports = createHook({
+  init(asyncId, type, triggerId, resource) {
+    if (!Number.isInteger(asyncId))
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'asyncId', 'integer');
+    if (typeof type !== 'string')
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'type', 'string');
```

Comment:
I added these lines based on the feedback from you and @jasnell:

> As for typechecking, I think the idea was to `if (…) throw errors.TypeError();` in JS land in any case (and I’m okay with that).

Having said that, I agree that if `type` is not a string then it's an internal Node.js error, and I am fine with crashing the process. I'll make the changes you are proposing.